subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -125,12 +190,18 @@ int dfu_target_mcuboot_done(bool successful)
 			LOG_ERR("boot_request_upgrade error %d", err);
 			return err;
 		}
-
 		LOG_INF("MCUBoot image upgrade scheduled. Reset the device to "
 			"apply");
 	} else {
 		LOG_INF("MCUBoot image upgrade aborted.");
 	}
+	/* Need to set bytes_written to 0 */
+	err = flash_img_init(&flash_img);
```

Comment:
Should you re-initialize it also in case `flash_img_buffered_write` or `boot_request_upgrade` failed?

Reply:
In my opinion it makes sens to reset the progress if either of those processes failed since it would mean you have a corrupt image.

Reply:
Ok, then you shouldn't `return` on line 189 and 195.

Reply:
Ok, then you shouldn't return on line 189 and 195.

Reply:
I think those returns could be valuable added it as a function so that it is called upon errors.